test/parallel/test-inspector-exit-worker-in-wait-for-connection2.mjs

```diff
@@ -0,0 +1,21 @@
+// Flags: --inspect=0
+import * as common from '../common/index.mjs';
+common.skipIfInspectorDisabled();
+
+import { isMainThread, Worker } from 'node:worker_threads';
+import { fileURLToPath } from 'node:url';
+import inspector from 'node:inspector';
+import assert from 'node:assert';
+
+// Ref: https://github.com/nodejs/node/issues/52467
+// - process.exit() should kill the main thread's process.
+if (isMainThread) {
+  new Worker(fileURLToPath(import.meta.url));
+  await new Promise((r) => setTimeout(r, 2_000));
+
+  process.on('exit', (status) => assert.strictEqual(status, 0));
+  process.exit();
```

Comment:
1. I suggest merging the test cases - start worker #1, terminate it, start worker #2, call `process.exit`
2. What would happen if the main thread runs to completion? Would the worker just stay hanging? I don't remember details, would the WebSocket server shut down when the main thread completes and there's no ongoing sessions?

I believe the broader question is if `inspector.waitForDebugger` even makes sense for workers. The way I remember target domain is you either specify autoattach so existing session would attach to newly created workers, or workers just start and sessions are attached whenever.

In the first case (autoattach) worker should only wait if there are connected sessions that asked to be attached to workers.
In the second case worker do not need to wait.

I apologize if I am not making sense, haven't worked on this in years...

Reply:
> I believe the broader question is if `inspector.waitForDebugger` even makes sense for workers. The way I remember target domain is you either specify autoattach so existing session would attach to newly created workers, or workers just start and sessions are attached whenever.

As far as I know, `inspector.waitForDebugger` is the only working way to (programatically) debug Workers. https://github.com/nodejs/node/issues/26609#issuecomment-1463605558

Reply:
Ack. Thank you for the explanations.